### PR TITLE
ci: Use rustc docker image for x86_64 and i686 linux

### PR DESCRIPTION
**Pull request by tesuji**
cc @kinnison , @alexcrichton 
~So the effort of improving CI time is zero for now~ (#1760)
The next PR might be setting up sccache.

### CODE REVIEW

File: ci/docker/i686-unknown-linux-gnu/Dockerfile
Comment (kinnison) [inline ci/docker/i686-unknown-linux-gnu/Dockerfile]: Is there a nicer, perhaps package-managery way we can have this work?

**Review by kinnison — APPROVED** [review]
Over-all I like this a lot.  It feels right to me to be using the same docker image sources everywhere.

@alexcrichton how do you feel about this?  I'm guessing positive?

(tentative approval, but I'd like at least one other docker-conversant person to have final approval)

File: ci/docker/i686-unknown-linux-gnu/Dockerfile
Comment (mati865) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: There is no package manager involved here.
Setting environment variable `CC=/rustroot/bin/gcc` should work.

Comment (tesuji) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: No, setting `CC` env doesn't seem to work. Maybe `cc` crate doesn't respect
`CC` env?

Comment (tesuji) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]:
```console
$ printenv CC CXX
clang
clang++
```

CC is set to clang and we still had the build error.

Comment (mati865) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: Ah, the target is explicitly configured: https://github.com/rust-lang/rustup.rs/blob/66c4d10871edc307865db031483f2bd799abbba3/ci/run.sh#L9
So it'd be `CC_x86_64-unknown-linux-gnu` for x84_64.

I don't know exact error but if it was the linker then `RUSTFLAGS="-C linker=gcc"` would help but this is getting a bit complex.

Comment (tesuji) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: Is `CC_x86_64-unknown-linux-gnu` standard or it's just used by cc crate?

Comment (tesuji) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: The error message is
```rust
     Running `rustc --crate-name semver_parser /src/target/cargo-home/registry/src/github.com-1ecc6299db9ec823/semver-parser-0.7.0/src/lib.rs --color always --crate-type lib --emit=dep-info,link -C opt-level=3 -C codegen-units=1 -C metadata=c7af5d73bcae4fa9 -C extra-filename=-c7af5d73bcae4fa9 --out-dir /src/target/release/deps -L dependency=/src/target/release/deps --cap-lints allow`
error: linker `cc` not found
  |
  = note: No such file or directory (os error 2)
```

Comment (mati865) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: > Is CC_x86_64-unknown-linux-gnu standard or it's just used by cc crate?

As far as I can tell it's used by `cc-rs` and Rust's bootstrap.

> The error message is

That's the linker error so `RUSTFLAGS="-C linker=gcc"` would work.

Comment (tesuji) [reply ci/docker/i686-unknown-linux-gnu/Dockerfile]: I still think about it: Why does rustc's docker image have no `cc` hard link?
It is common that `cc` hard link to `gcc` on Linux distro.

**Comment by tesuji** [comment]
Blocked by rust-lang/rust#60366

**Review by alexcrichton — APPROVED** [review]

**Comment by tesuji** [comment]
Ok. Ready to merge :tada:

**Pull request merged**
